pyteomics/proforma.py

```diff
@@ -2592,7 +2592,10 @@ def composition(self, include_charge=False, aa_comp=None):
                 comp += aa_comp[aa]
                 for tag in tags or []:
                     comp += tag.composition
-            for tag in chain(self.fixed_modifications, self.labile_modifications, self.unlocalized_modifications):
+                for rule in self.fixed_modifications:
+                    if rule.is_valid(aa, self.n_term, self.c_term):
```

Comment:
`rule.is_valid` has the signature `f(amino_acid: str, n_term: bool, c_term: bool)` where `n_term` and `c_term` indicate that the amino acid is at the N-terminal or C-terminal respectively. 

The for loop in `mass` uses the index to determine whether the current position is the N or C terminal:
https://github.com/levitsky/pyteomics/blob/3b2d87d00de3ea7a8002d6f2171b9748a32db28f/pyteomics/proforma.py#L2356-L2369

Specifically lines 2364 and 2365 tie these together.

Reply:
Oh, I am obviously not in the sharpest form working on this. Thanks for catching.